code/CMSProfileController.php

```diff
@@ -67,11 +67,21 @@ public function canView($member = null)
             return false;
         }
 
-        // Check they can access the CMS and that they are trying to edit themselves
-        if (Permission::checkMember($member, "CMS_ACCESS")
-            && $member->ID === Security::getCurrentUser()->ID
-        ) {
-            return true;
+        // Check they are trying to edit themselves
+        if ($member->ID === Security::getCurrentUser()->ID) {
+            $codes = $this->getRequiredPermissions();
+
+            // allow explicit TRUE
+            if ($codes === true) {
+                return true;
+            }
+
+            // Check they can access self::$required_permission_codes
+            foreach ((array)$codes as $code) {
```

Comment:
I'm not sure I like this style of forcing an array. I think that `getRequiredPermissions` should return a consistent API which is always an array.

Reply:
I'm not sure it's always an array. I think it could return an array, a string, or a bool; https://github.com/silverstripe/silverstripe-admin/blob/1/code/LeftAndMain.php#L578 ?

Looking at this again now; I also don't like the style of forcing an array. Would you be ok with a simple `is_array` test?

Reply:
Sure, us_array would be fine

Reply:
Actually `Permission::checkMember()` will accept an array so I've tidied this up.